p3/plot/_cascade.py

```diff
@@ -376,18 +392,42 @@ def cascade(df, eff=None, **kwargs):
         handles=app_handles, loc="upper left", bbox_to_anchor=(1, 1), ncol=1
     )
 
-    # Attach the platform legend to the bottom of the figure
+    # Attach the platform legend to the position requested by kwargs
     legend_helper = _PlatformLegendHandler(plat_colors, plat_labels)
     plat_handles = [
         mpatches.Patch(color=plat_colors[p], label=p) for p in platforms
     ]
-    fig.legend(
-        handles=plat_handles,
-        loc="upper center",
-        bbox_to_anchor=(0.5, 0.0),
-        handler_map={mpatches.Patch: legend_helper},
-        handlelength=1.0,
-        ncols=4,
-    )
+    if not isinstance(plat_legend_ncols, int):
```

Comment:
Might be a matter of style (or recent mood), but I would suggest `assert` for the next three errors being raised (since you're actually testing for a type).

```python
assert isinstance(plat_legend_ncols, int), "plat_legend_ncols must be integer"
assert plat_legend_loc in ["north", "east", "south", "west", "off"], f"Invalid plat_legend_loc value; expected 'north', 'east', 'south', 'west', or 'off', got {plat_legend_loc}."
```

Reply:
`assert` is less code, but I'd rather not make this change here.  All the other checks (in other files) raise `ValueErrors` for these sorts of checks.  We should discuss this separately, and decide if we want to change to `assert` everywhere.

Reply:
That's fair - what I meant was more that `assert` made more sense semantically than a conditional statement, hence it's more style than anything :)